src/FlipMove.js

```diff
@@ -624,6 +624,10 @@ class FlipMove extends Component<void, ConvertedProps, FlipMoveState> {
 
   removeChildData(key: string): void {
     delete this.childrenData[key];
+    this.setState(prevState => ({
+      ...prevState,
```

Comment:
I don't believe this line is necessary; I know it isn't for the traditional object `setState` call, but I believe the functional version is the same. It merges in what you return with the existing state.

Reply:
> prevState is a reference to the previous state. It should not be directly mutated. Instead, changes should be represented by building a new object based on the input from prevState and props
https://facebook.github.io/react/docs/react-component.html#setstate

im actually pretty sure that whatever you return overwrites anything in the store. how would you remove entries from the store otherwise? at the very least, it seems to be encouraged by facebook (and immutability design patterns) to return an entire state object